File: src/backend/cs/StudyRoomSystem/StudyRoomSystem.Core/Structs/Entity/Complaint.cs
using System.Text.Json.Serialization;

namespace StudyRoomSystem.Core.Structs.Entity;

[JsonConverter(typeof(JsonStringEnumConverter<ComplaintStateEnum>))]
public enum ComplaintStateEnum
{
    无, 已发起, 已处理, 已关闭
}

/// <summary>
/// 投诉记录
/// </summary>
public class Complaint
{
    public Guid Id { get; set; } = Ulid.NewUlid().ToGuid();
    public required Guid SendUserId { get; set; }
    public required Guid SeatId { get; set; }
    public required ComplaintStateEnum State { get; set; }
    public required string Type { get; set; }
    public required string SendContent { get; set; }
    public DateTime? TargetTime { get; set; }
    public DateTime CreateTime { get; set; } = DateTime.UtcNow;
    public DateTime? HandleTime { get; set; }
    public Guid? HandleUserId { get; set; }
    public string? HandleContent { get; set; }

    public virtual User SendUser { get; set; } = null!;
    public virtual Seat Seat { get; set; } = null!;
    public virtual User? HandleUser { get; set; }
}

File: src/frontend/web/study-room-system/src/lib/types/Complaint.ts
import type { Seat } from "./Seat";
import type { User } from "./User";

export type ComplaintState = "无" | "已发起" | "已处理" | "已关闭";

export interface Complaint {
  id: string;
  sendUserId: string;
  seatId: string;
  state: ComplaintState;
  type: string;
  sendContent: string;
  targetTime: string | null;
  createTime: string;
  handleTime: string | null;
  handleUserId: string | null;
  handleContent: string | null;
  sendUser?: User | null;
  seat?: Seat | null;
  handleUser?: User | null;
}

export interface ComplaintPage {
  total: number;
  page: number;
  pageSize: number;
  items: Complaint[];
}

export function localizeComplaintState(state: ComplaintState) {
  return state;
}
